daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/ContractStateMachine.scala

```diff
@@ -235,7 +235,11 @@ class ContractStateMachine[Nid](mode: ContractKeyUniquenessMode) {
           // Check if we have a cached key input.
           globalKeyInputs.get(gkey) match {
             case Some(keyInput) =>
-              val keyMapping = keyInput.toKeyMapping
+              val keyMapping = keyInput match {
```

Comment:
I somewhat wonder if this should be a method on the global contract state rather than some logic but don’t feel strongly about this either way.

Reply:
For now I will let it there, as it is not use anywhere else.